HorizonApache/src/test/java/com/hubspot/horizon/apache/ApacheHttpClientTest.java

```diff
@@ -52,6 +52,18 @@ public void itWorksWithHttp() {
     assertThat(response).hasStatusCode(200).hasBody("").hasRetries(0);
   }
 
+  @Test(expected = Exception.class)
```

Comment:
Can we do assertThatThrownBy and then actually make assertions about the type and message on the exception? Such as checking that the cause is `GeneralSecurityException`?

There are multiple places in the PR where this would be useful.